### ISSUE TITLE
rename and delete user

### ISSUE THREAD

**Issue by nazoking** [gitbucket/gitbucket#391]
I want to correspond to the misspelling at the time of registration.

**Comment by gotnix** [comment]
+1
I tried version 3.6, without this feature.

**Comment by skironDotNet** [comment]
this is actually easy to hack via Data export/import

**Comment by shinsenter** [comment]
+1 for user deletion.

**Comment by aadrian** [comment]
@skironDotNet 
> "user name is the PK"

That the user name is "also a primary" key is not the main problem (it should be however only unique (and natural key https://en.wikipedia.org/wiki/Natural_key ) ).
The main problem is the **missing of a numeric ID** for the row in the `account` table !

That means that "updating" the user name is a complex task since the user name string is "hardcoded" everywhere so it's not enough to update the only the `account` table.

**Comment by skironDotNet** [comment]
@aadrian Yes, I was not aware of the DB design until I started purging "deleted" users.
So this task #391 is really significant amount of work
1. need to create DB migration script to (abstract SQL) `Alter account add column Id auto identity` 
  - add `user_id` column to any table that has "hardcoded" user name
  - assign user_id value based on account.id where username = x like update with JOIN
  - delete "hardcoded" user name column from every table except account
2. Update all the code (data access layer) that if currently joins on undename, to use proper PK/FK approach `natural_key`

Now that we have migration to update existing DB for the users and proper DAL

3. Admin UI page to Rename and Delete users
   - soft delete means "disbale" user do not delete repositories 
   - hard delete means remove everything about that user including repositories 
  
  Since I'm not scala, java developer I can't help with that, maybe migration script I could do but then the owner would have to approve up front he will merge the PR otherwise it's just wasted time on my side
